network/endpoint_linux.go

```diff
@@ -353,15 +350,15 @@ func deleteRoutes(nl netlink.NetlinkInterface, netioshim netio.NetIOInterface, i
 			}
 
 			ifIndex = devIf.Index
-		} else {
+		} else if interfaceName != "" {
```

Comment:
Was this failing in a case where `interfaceName` is empty ?
Then should we add another else and log the error or something.

Reply:
it would not fail but skip deleting the route if interface not provided but the requirement is to delete route even if interface name not provided